### ISSUE TITLE
EGL_DISPMANX_WINDOW_T don't declared.

### ISSUE THREAD

**Issue by Xnork** [libsdl-org/SDL#5132]
[ 32%] Building C object CMakeFiles/SDL2-static.dir/src/audio/sndio/SDL_sndioaudio.c.o
[ 32%] Building C object CMakeFiles/SDL2-static.dir/src/video/raspberry/SDL_rpievents.c.o
In file included from /home/pi/Downloads/SDL2-2.0.18/src/video/raspberry/SDL_rpievents.c:29:
/home/pi/Downloads/SDL2-2.0.18/src/video/raspberry/SDL_rpivideo.h:47:5: error: unknown type name ‘EGL_DISPMANX_WINDOW_T’
   47 |     EGL_DISPMANX_WINDOW_T dispman_window;
      |     ^~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/SDL2-static.dir/build.make:1753: CMakeFiles/SDL2-static.dir/src/video/raspberry/SDL_rpievents.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:127: CMakeFiles/SDL2-static.dir/all] Error 2
make: *** [Makefile:149: all] Error 2

**Comment by slouken** [comment]
Does it work if you use configure and make?

**Comment by cmitu** [comment]
I think it works correctly when using `configure` / `make`, but not using `cmake`.
The error is probably happening on (the new) Raspbian Bullseye, where the Dispmanx headers/libraries are no longer used and are not present anymore (they used to be in `libraspberrrypi-dev` on Buster / older Raspbian versions).

```
./configure # no args
...
checking for bcm_host brcmegl... no
checking for Raspberry Pi 2/3... no
...
SDL2 Configure Summary:
Building Shared Libraries
Building Static Libraries
Enabled modules : atomic audio video render events joystick haptic hidapi sensor power filesystem threads timers file loadso cpuinfo assembly
Assembly Math   :
Audio drivers   : disk dummy oss alsa(dynamic) pulse(dynamic) sndio(dynamic)
Video drivers   : dummy x11(dynamic) kmsdrm(dynamic) opengl opengl_es1 opengl_es2 vulkan wayland(dynamic)
X11 libraries   : xcursor xdbe xinerama xinput2 xinput2_multitouch xfixes xrandr xscrnsaver xshape xvidmode
Input drivers   : linuxev linuxkd
Enable virtual joystick APIs : YES
```

Wheres using `cmake`:
```
...
-- Checking for modules 'bcm_host;brcmegl'
--   No package 'brcmegl' found
-- Performing Test HAVE_RPI
-- Performing Test HAVE_RPI - Success
...

--   SDL_RPATH                   (Wanted: ON): ON
--   SDL_RPI                     (Wanted: ON): ON
--   SDL_SNDIO                   (Wanted: ON): ON
--   SDL_SNDIO_SHARED            (Wanted: ON): ON
....
```

and sure enough, there's a build error:
```
[ 60%] Building C object CMakeFiles/SDL2-static.dir/src/video/raspberry/SDL_rpimouse.c.o
In file included from /home/pi/src/SDL-release-2.0.18/src/video/raspberry/SDL_rpievents.c:29:
/home/pi/src/SDL-release-2.0.18/src/video/raspberry/SDL_rpivideo.h:47:5: error: unknown type name ‘EGL_DISPMANX_WINDOW_T’
   47 |     EGL_DISPMANX_WINDOW_T dispman_window;
      |     ^~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/SDL2-static.dir/build.make:1753: CMakeFiles/SDL2-static.dir/src/video/raspberry/SDL_rpievents.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 60%] Building C object CMakeFiles/SDL2.dir/src/video/SDL_vulkan_utils.c.o
In file included from /home/pi/src/SDL-release-2.0.18/src/video/raspberry/SDL_rpimouse.c:28:
/home/pi/src/SDL-release-2.0.18/src/video/raspberry/SDL_rpivideo.h:47:5: error: unknown type name ‘EGL_DISPMANX_WINDOW_T’
   47 |     EGL_DISPMANX_WINDOW_T dispman_window;
      |     ^~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/SDL2-static.dir/build.make:1766: CMakeFiles/SDL2-static.dir/src/video/raspberry/SDL_rpimouse.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:127: CMakeFiles/SDL2-static.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
...
```

So it seems only the `cmake` build path has this error.

**Comment by icculus** [comment]
Going to try to sneak this into 2.0.20, but it might slip depending on how the next few days go.
